tools/killsnoop.bt

```diff
@@ -31,9 +31,9 @@ tracepoint:syscalls:sys_enter_kill
 tracepoint:syscalls:sys_exit_kill
 /@tpid[tid]/
 {
-	time("%H:%M:%S  ");
-	printf("%-6d %-16s %-4d %-6d %d\n", pid, comm, @tsig[tid], @tpid[tid],
-	    args.ret);
+	printf("%-15s %7d %-16s %4d %6d %6d\n",
+		strftime("%H:%M:%S.%f", nsecs),
```

Comment:
Are the nanoseconds really necessary? Do you have use-cases where the current granularity is not sufficient?

Reply:
The bigger problem that output of `time` and `printf` can be interleaved.

It debugging of real time issues one second is very rough.  When you have another log with hundreds of messages per second, microseconds are very helpful in finding correlations.

[strftime](https://github.com/iovisor/bpftrace/blob/master/docs/reference_guide.md#25-strftime-formatted-timestamp)  receives integer nanoseconds and provides decimal seconds only in microseconds (six digits). It can't be changed.

Reply:
Ok, let's use microseconds, then.